stellargraph/data/explorer.py

```diff
@@ -51,42 +49,18 @@ def __init__(self, graph, graph_schema=None, seed=None):
 
         # We require a StellarGraph for this
         if not isinstance(graph, StellarGraphBase):
-            raise TypeError(
-                "Graph must be a StellarGraph or StellarDiGraph to use heterogeneous sampling."
-            )
+            raise TypeError("Graph must be a StellarGraph or StellarDiGraph.")
 
         if not graph_schema:
             self.graph_schema = self.graph.create_graph_schema(create_type_maps=True)
         else:
             self.graph_schema = graph_schema
 
-        if self.graph_schema is not None and type(self.graph_schema) is not GraphSchema:
-            raise ValueError(
-                "({}) The parameter graph_schema should be either None or of type GraphSchema.".format(
-                    type(self).__name__
-                )
+        if type(self.graph_schema) is not GraphSchema:
+            self._raise_error(
+                "The parameter graph_schema should be either None or of type GraphSchema."
             )
 
-        # Create a dict of adjacency lists per edge type, for faster neighbour sampling from graph in SampledHeteroBFS:
```

Comment:
any reason for moving this?

Reply:
Moved it into its own base class, since it is only used in one place.

Reply:
But this required creating a new base class for heterogeneous walks. I think this is not necessary, so I'd prefer to have only one base class here. Once the StellarGraph class is refactored, we'll either get rid of explicit use of adjacency lists per edge type here, or will use them in all descendent walk classes.